libdnf5-plugins/actions/actions.cpp

```diff
@@ -844,7 +854,777 @@ void Actions::process_command_output_line(std::string_view line) {
+                        } else if (param_key == "IGNORE_REGULAR_USER_EXCLUDES") {
+                            query_flags = query_flags | libdnf5::sack::ExcludeFlags::IGNORE_REGULAR_USER_EXCLUDES;
+                        } else {
+                            throw JsonRequestError(fmt::format("Bad key \"{}\" for params", param_key));
+                        }
+                    }
+                }
+
+                auto * jfilters = get_any_object_or_null(jargs, "filters");
+                if (jfilters) {
+                    get_array(jargs, "filters");  // check, must be array
+                }
+
+                enum class Direction { IN, OUT, ALL } edirection = Direction::ALL;
+                if (jfilters) {
+                    for (size_t idx = 0; idx < json_object_array_length(jfilters); ++idx) {
+                        auto * jfilter = json_object_array_get_idx(jfilters, idx);
+                        auto filter_key = get_string_view(jfilter, "key");
+                        if (filter_key == "direction") {
+                            auto direction = get_string_view(jfilter, "value");
+                            if (direction == "IN") {
+                                edirection = Direction::IN;
+                            } else if (direction == "OUT") {
+                                edirection = Direction::OUT;
+                            } else {
+                                throw JsonRequestError(
+                                    fmt::format("Bad \"{}\" value for \"direction\" filter", direction));
+                            }
+                            break;
+                        }
+                    }
+                }
+
+                auto query = domain == "packages"
+                                 ? libdnf5::rpm::PackageQuery(get_base(), query_flags)
+                                 : (edirection == Direction::IN
+                                        ? *in_full_query
+                                        : (edirection == Direction::OUT ? *out_full_query : *all_full_query));
+
+                if (jfilters) {
+                    for (size_t idx = 0; idx < json_object_array_length(jfilters); ++idx) {
+                        auto * jfilter = json_object_array_get_idx(jfilters, idx);
+                        const auto filter_key = get_string_view(jfilter, "key");
+                        auto * jvalue = get_any_object_or_null(jfilter, "value");
+                        const auto value = jvalue ? std::string(get_string_view(jvalue)) : "";
+                        auto * joperator = get_any_object_or_null(jfilter, "operator");
+                        const auto oper = joperator ? cmp_operator_from_string(get_string_view(joperator))
+                                                    : libdnf5::sack::QueryCmp::EQ;
+                        if (filter_key == "direction") {
+                        } else if (filter_key == "name") {
```

Comment:
Minor suggestion but perhaps we could add a comment here saying that the `direction` filtering was already done above.

Reply:
I added comments.